org.eclipse.scanning.api/src/org/eclipse/scanning/api/scan/ScanEstimator.java

```diff
@@ -169,14 +184,21 @@ public int getRank() {
 			}
 		}
 		
-		this.shape = new int[getRank()];
+		final int scanRank = getRank();
+		// special case for scans of rank 0 - i.e. acquire scans
+		if (scanRank == 0) return new int[0]; 
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)